src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java

```diff
@@ -75,7 +75,12 @@ public void publish(final ChecksDetails details) {
 
             String apiUri = null;
             if (credentials instanceof GitHubAppCredentials) {
-                apiUri = ((GitHubAppCredentials) credentials).getApiUri();
+                final var gitHubAppCredentials = (GitHubAppCredentials) credentials;
+                apiUri = gitHubAppCredentials.getApiUri();
+                if (context instanceof GitHubSCMSourceChecksContext) {
+                    final var gitHubSCMSourceChecksContext = (GitHubSCMSourceChecksContext) context;
+                    credentials = gitHubAppCredentials.withOwner(gitHubSCMSourceChecksContext.getOwner());
```

Comment:
I think you could add a test to https://github.com/jenkinsci/github-checks-plugin/blob/c8db0623c4ea8d5938ae7a29a23f821298143cbf/src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java#L77 which would verify the owner is passed along.

In the existing test the owner is set on the SCM context but not on the credentials